litellm/llms/mistral/mistral_chat_transformation.py

```diff
@@ -287,11 +290,11 @@ def _handle_name_in_message(cls, message: AllMessageValues) -> AllMessageValues:
         """
         Mistral API only supports `name` in tool messages
 
-        If role == tool, then we keep `name`
+        If role == tool, then we keep `name` if it's not an empty string
         Otherwise, we drop `name`
         """
         _name = message.get("name")  # type: ignore
-        if _name is not None and message["role"] != "tool":
+        if _name is not None and (message["role"] != "tool" or (isinstance(_name, str) and len(_name.strip()) == 0)):
```

Comment:
`(message["role"] != "tool" or (isinstance(_name, str) and len(_name.strip()) == 0)):` 

Can this be a well named helper method

Reply:
Having a helper method for a helper method seemed like it would add some complexity, so instead of adding another helper method, I refactored it to make the logic cleaner and added comments to explain the logic. Hope that is what you were looking for 🎉